src/include/sof/drivers/sdma.h

```diff
@@ -78,6 +78,7 @@
 #define SDMA_DEFPRI		4
 #define SDMA_MAXPRI		7
 
+#define SDMA_CHANNELS_COUNT	32
```

Comment:
Why don't we use the .count field?

Reply:
we could, but this is more like MAX HW channels and .count field says how many channels we actually want to use.

Reply:
Are we confident that all SDMA implementations will have 32 channels in the future?

Reply:
It's safer to use dma->plat_data.channels. I don't think we'll put it to something else anyway -- we could probably create another limit in channel_get per platform.

This driver works standalone on the full device and thus should need to know how many channels are in-hardware anyway. We don't have a driver that works as a slave yet, with the actual allocation of channels and managing of them done on the ARM side.

Reply:
ok, no problem. will fix that.